xio-core/src/main/java/com/xjeffrose/xio/config/ConfigReloader.java

```diff
@@ -58,35 +60,28 @@ public ConfigReloader(ScheduledExecutorService executor, Function<Config, T> fac
     this.updater = null;
   }
 
-  private Config parse(Reader reader) {
-    Config config = ConfigFactory.parseReader(reader);
+  private Config parse(File file) {
+    Config config = ConfigFactory.parseFile(file);
     return ConfigFactory.load(config);
   }
 
-  private Meta<T> load(File file) {
+  private Meta<T> load(String filePath) {
     final MessageDigest digest;
     try {
       digest = MessageDigest.getInstance("SHA-256");
     } catch (Exception e) {
-      throw new IllegalStateException(
-          "Couldn't load config from file '" + file.getAbsolutePath() + "'", e);
+      throw new IllegalStateException("Couldn't load config from file '" + filePath + "'", e);
     }
-
-    try (FileInputStream fileInputStream = new FileInputStream(file);
-        DigestInputStream digestInputStream = new DigestInputStream(fileInputStream, digest);
-        InputStreamReader reader = new InputStreamReader(digestInputStream)) {
-      T value = factory.apply(parse(reader));
+    try {
```

Comment:
try with resource will auto close the resources. regular try doesn't. Is that an issue here since you are changing it from try with resources to just a regular try?

Reply:
will double check if needed

Reply:
file does not implement closable so =p will have to close manually